### PR TITLE
Feature/63/overlay indoor path

### PR DESCRIPTION
**Pull request by domludera**
This pull request will finally merge the indoor path functionality into the App.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -103,5 +103,11 @@ dependencies {
     androidTestImplementation 'androidx.test:rules:1.2.0'
     androidTestImplementation 'androidx.test:runner:1.2.0'
     androidTestImplementation 'androidx.test.espresso:espresso-idling-resource:3.2.0'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
+    androidTestImplementation "com.android.support.test.espresso:espresso-intents:3.2.0"
+    implementation project(path: ':a-star')
+//    implementation 'com.googlecode.json-simple:json-simple:1.1.1'
```
[the remark this answers is not in the document]
Comment (domludera) [reply app/build.gradle]: done

File: app/src/main/java/com/conupods/IndoorMaps/View/IndoorNavigation.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            System.err.println(e);
+        }
+
+        return in;
+    }
+
+
+    public boolean[][] createBinaryGrid(String filePath) {
+
+        BufferedReader in = new BufferedReader(getInputStreamReader(filePath));
```
[the remark this answers is not in the document]
Comment (domludera) [reply app/src/main/java/com/conupods/IndoorMaps/View/IndoorNavigation.java]: done

File: app/src/main/java/com/conupods/IndoorMaps/View/IndoorPath.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        AStar aStar = new AStar();
+        aStar.mMetadataFilePath = "json/Metadata.json";
+
+        IndoorNavigation indoorNavigation = new IndoorNavigation();
+        InputStreamReader in = indoorNavigation.getInputStreamReader(aStar.mMetadataFilePath);
+
+        Destination start = aStar.setDestFromString(startPoint);
+        Destination end = aStar.setDestFromString(endPoint);
+
+        Edges[] startEnd = aStar.getDictFromJSON(start, end, in);
```
[the remark this answers is not in the document]
Comment (domludera) [reply app/src/main/java/com/conupods/IndoorMaps/View/IndoorPath.java]: done

File: library/a-star/src/test/java/AStarSystemTest.java [hunk 4]
```diff
@@ -1,51 +1,97 @@
...
-        Spot walk = algo(grid, x1, y1, x2, y2);
+        boolean[][] binGrid = null;
+
+        try {
+            binGrid = createBinaryGrid("metadata/gridfile/h9");
+        } catch (IOException e) {
+            System.err.println(e);
+        }
+
+
```
[the remark this answers is not in the document]
Comment (domludera) [reply library/a-star/src/test/java/AStarSystemTest.java]: done

File: library/a-star/src/main/java/Main.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
+import astar.AStar;
```
[the remark this answers is not in the document]
Comment (domludera) [reply library/a-star/src/main/java/Main.java]: I don't, forgot to delete. Thanks for noticing

**Comment by MinaSelim** [comment]
I merged this PR through git due to conflict issues

**Pull request closed without merging**